File: Exemplos/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/SAT/cancelar_ultima_venda.dart
import 'package:intent_digital_hub/IntentDigitalHubService/SAT/sat_command.dart';

class CancelarUltimaVenda extends SatCommand {
  final int _numSessao;
  final String _codAtivacao;
  final String _numeroCFe;
  final String _dadosCancelamento;

  CancelarUltimaVenda(this._numSessao, this._codAtivacao, this._numeroCFe,
      this._dadosCancelamento)
      : super('CancelarUltimaVenda');

  @override
  functionParameters() {
    return '"numSessao"'
        ':'
        '$_numSessao'
        ','
        '"codAtivacao"'
        ':'
        '"$_codAtivacao"'
        ','
        '"numeroCFe"'
        ':'
        '"$_numeroCFe"'
        ','
        '"dadosCancelamento"'
        ':'
        '"'
        '$_dadosCancelamento'
        '"';
  }
}

File: Exemplos/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/SAT/CancelarUltimaVenda.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.SAT

import com.elgin.intent_digital_hub.IntentDigitalHubService.SAT.SatCommand

class CancelarUltimaVenda(
    private val numSessao: Int,
    private val codAtivacao: String,
    private val numeroCFe: String,
    private val dadosCancelamento: String
) : SatCommand("CancelarUltimaVenda") {
    override fun functionParameters(): String {
        return "\"numSessao\"" + ":" + numSessao + "," +
                "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"" + "," +
                "\"numeroCFe\"" + ":" + "\"" + numeroCFe + "\"" + "," +
                "\"dadosCancelamento\"" + ":" + "\"" + dadosCancelamento + "\""
    }
}